src/main/java/kr/kro/deom/domain/exchange/service/StampExchangeService.java

```diff
@@ -226,15 +226,14 @@ private void performStampExchange(StampExchange exchange, Long responderId) {
         deductOrThrow(creatorId, sourceStoreId, sourceAmount);
         deductOrThrow(responderId, targetStoreId, targetAmount);
 
-        int updated1 = myStampRepository.addStampAmount(creatorId, targetStoreId, targetAmount);
-        if (updated1 == 0) {
-            MyStamp myStamp = new MyStamp(creatorId, targetStoreId, targetAmount);
-            myStampRepository.save(myStamp);
-        }
-        int updated2 = myStampRepository.addStampAmount(responderId, sourceStoreId, sourceAmount);
-        if (updated2 == 0) {
-            MyStamp myStamp = new MyStamp(responderId, sourceStoreId, sourceAmount);
-            myStampRepository.save(myStamp);
+
+        int updated1 = myStampRepository.updateStampAmount(creatorId, targetStoreId, targetAmount);
+        int updated2 =
+                myStampRepository.updateStampAmount(responderId, sourceStoreId, sourceAmount);
+
+        if (updated1 != 1 || updated2 != 1) {
+            throw new StampExchangeException(CommonErrorCode.STAMP_EXCHANGE_EXECUTION_FAILED);
+
```

Comment:
```suggestion

        int updated1 = myStampRepository.addStampAmount(creatorId, targetStoreId, targetAmount);
        if (updated1 == 0) {
            MyStamp myStamp = new MyStamp(creatorId, targetStoreId, targetAmount);
            myStampRepository.save(myStamp);
        }
        int updated2 = myStampRepository.addStampAmount(responderId, sourceStoreId, sourceAmount);
        if (updated2 == 0) {
            MyStamp myStamp = new MyStamp(responderId, sourceStoreId, sourceAmount);
            myStampRepository.save(myStamp);

```

이 부분은 바뀌면 안 됩니다. 지금 배포되어있는 로직이 맞는데 혹시 어떤 이유로 변경하셨나요?